parser/src/main/kotlin/tools/samt/parser/Nodes.kt

```diff
@@ -16,6 +17,11 @@ sealed class StatementNode(
     location: Location,
 ) : Node(location)
 
+sealed class NamedDeclarationNode(
+    val name: IdentifierNode,
```

Comment:
What's the purpose of keeping the IdentifierNode around? Why not just store a single string?

Reply:
It's in the parser nodes, where we might want to highlight the name of a given thing for diagnostics
I simply added a new common base node to reduce code copy-paste